libsql-server/src/namespace/mod.rs

```diff
@@ -473,7 +474,7 @@ impl<M: MakeNamespace> NamespaceStore<M> {
                 &self.inner.metadata,
             )
             .await?;
-
+        self.inner.metadata.remove(&namespace)?;
```

Comment:
I would remove the entry from the state before deallocating the in memory resources, because you don't hold a lock to it. What could happen here is:
- deallocate the namespace
- a request tries to allocate it again, it still exist in the store, so we can reallocate
- we perform the metadata deletion

Reply:
Apparently changing the order causes the destroy namespace test to fail. I need to check out why.

Reply:
The reason why switching the order didn't work was that we first removed db config from MetaStore, but then in `PrimaryNamespace::destroy` we access it by get-or-create semantics, therefore reinserting it again. 

I've changed the behaviour, so not `PrimayNamespace::destroy` accesses the DB config by remove-and-return semantics from MetaStore.